actors/builtin/miner/miner_actor.go

```diff
@@ -3108,3 +3125,114 @@ func checkPeerInfo(rt Runtime, peerID abi.PeerID, multiaddrs []abi.Multiaddrs) {
+	}
+
+	if params.NewQuota.LessThanEqual(big.NewInt(0)) {
+		rt.Abortf(exitcode.ErrIllegalArgument, "beneficial quota (%s) should bigger than zero", params.NewQuota)
+	}
+}
+
+// ChangeBeneficiary proposal/approve beneficiary change
+func (a Actor) ChangeBeneficiary(rt Runtime, params *ChangeBeneficiaryParams) *abi.EmptyValue {
+	if params == nil {
+		rt.Abortf(exitcode.ErrIllegalArgument, "empty ChangeBeneficiary Paramenters")
+	}
+	if params.NewBeneficiary.Protocol() != addr.ID {
+		rt.Abortf(exitcode.ErrIllegalArgument, "NewBeneficiary address must be an ID address")
+	}
+
+	var st State
+	rt.StateTransaction(&st, func() {
+		info := getMinerInfo(rt, &st)
+
+		// This is a ChangeBeneficiary proposal when the caller is Owner
+		if rt.Caller() == info.Owner {
+			rt.ValidateImmediateCallerIs(info.Owner)
+			if params.NewBeneficiary != info.Owner {
+				// No need to check others when the newBeneficiary is set back to owner
+				checkNewBeneficialParam(rt, params)
+			}
+			if info.Beneficiary != info.Owner && info.BeneficiaryInfo.Effective(rt.CurrEpoch()) {
+				info.PendingBeneficiaryInfo = &PendingBeneficiaryChange{
+					NewBeneficiary: params.NewBeneficiary,
+					NewQuota:       params.NewQuota,
+					NewExpiration:  params.NewExpiration,
+					NextApprover:   info.Beneficiary,
+				}
+			} else if params.NewBeneficiary != info.Owner {
+				info.PendingBeneficiaryInfo = &PendingBeneficiaryChange{
+					NewBeneficiary: params.NewBeneficiary,
+					NewQuota:       params.NewQuota,
+					NewExpiration:  params.NewExpiration,
+					NextApprover:   params.NewBeneficiary,
+				}
+			} else {
+				info.Beneficiary = params.NewBeneficiary
+				info.BeneficiaryInfo = BeneficiaryInfo{
+					Quota:      big.Zero(),
+					Expiration: 0,
+					UsedQuota:  big.Zero(),
+				}
+				info.PendingBeneficiaryInfo = nil
+			}
+		} else {
+			// Non-owner calls ChangeBeneficiary is to approve a pending proposal
+			builtin.RequireState(rt, info.PendingBeneficiaryInfo != nil, "No changeBeneficiary proposal exists")
+			rt.ValidateImmediateCallerIs(info.PendingBeneficiaryInfo.NextApprover)
+
+			builtin.RequireParam(rt, info.PendingBeneficiaryInfo.NewBeneficiary == params.NewBeneficiary, "new beneficiary address must be equal expect %s, but got %s", info.PendingBeneficiaryInfo.NewBeneficiary, params.NewBeneficiary)
+			builtin.RequireParam(rt, info.PendingBeneficiaryInfo.NewQuota.Equals(params.NewQuota), "new beneficiary quota must be equal expect %s, but got %s", info.PendingBeneficiaryInfo.NewQuota, params.NewQuota)
+			builtin.RequireParam(rt, info.PendingBeneficiaryInfo.NewExpiration == params.NewExpiration, "new beneficiary expiredate must be equal expect %s, but got %s", info.PendingBeneficiaryInfo.NewExpiration, params.NewExpiration)
+
+			if params.NewBeneficiary != rt.Caller() && params.NewBeneficiary != info.Owner {
```

Comment:
I think `params.NewBeneficiary == info.Owner` is impossible, so this check is confusing.

Explain this better. "If the caller is not the new beneficiary, the caller must be the old beneficiary, and so the proposal still needs approval by the new one". Given that explanation, should we check the caller really is the old beneficiary? This may get simpler if the approval chain is more flexible.